src/bootstrap.coffee

```diff
@@ -179,38 +179,51 @@ hasDeviceApiKeySupport = (osVersion) ->
 		console.error('Unable to determine if device has deviceApiKey support', err, err.stack)
 		false
 
+exchangeKeyAndUpdateConfig = ->
+	# Only do a key exchange and delete the provisioning key if we're on a Resin OS version
+	# that supports using the deviceApiKey (2.0.2 and above)
+	# or if we're in a non-Resin OS (which is assumed to be updated enough).
+	# Otherwise VPN and other host services that use an API key will break.
+	#
+	# In other cases, we make the apiKey equal the deviceApiKey instead.
+	osRelease.getOSVersion(config.hostOSVersionPath)
+	.then (osVersion) ->
+		hasSupport = hasDeviceApiKeySupport(osVersion)
```

Comment:
Since you're at it, would it make sense to use [resin-semver](https://www.npmjs.com/package/resin-semver) in `hasDeviceApiKeySupport()` to compare the resinOS versions?

Reply:
@lekkas will do, but in a separate PR